### PR TITLE
MRA-15: Refactor favorites logic

### PR DESCRIPTION
**Pull request by Dabble362**
add favorites.js controller, removed redundant logic in recipe.js controllers, refactor favorites route in recipes route

### CODE REVIEW

File: controllers/favorites.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  getFavorites: async (req, res) => {
+    const currentPage = "favorites";
+    const skip =
+      parseInt(req.query.skip || "0", 10) <= 0
+        ? 0
+        : parseInt(req.query.skip, 10);
+    const limit = req.query.limit || 4;
+    console.log(currentPage);
+    try {
+      const totalFavorites = await Favorite.find({
```
Comment (jduffey) [inline controllers/favorites.js:14]: Very minor formatting suggestion -- these lines could be made to be slightly more symmetrical like this:
```
const totalFavorites = await Favorite.find({ user: req.user.id })
  .countDocuments();
const recipes = await Favorite.find({ user: req.user.id })
  .populate("recipe")
  .skip(skip)
  .limit(limit);```

**Review by jduffey — CHANGES_REQUESTED** [review]
Two minor changes requested.

File: controllers/favorites.js (hunk 1 above)
Comment (Dabble362) [reply controllers/favorites.js:14]: Changed! Making the commits now!

**Review by jduffey — APPROVED** [review]
:shipit:

**Pull request merged**